### PR TITLE
Fix button visibility issue on dark theme

### LINKED ISSUE #38113: Dark theme: Base button text is not visible
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

The text of buttons using only the `.btn` class is not visible when using the dark theme.

Screenshot from 2023-02-24 14-55-56

Looking at the code, there's nothing that redefines `--bs-btn-color` back to the body color when switching themes, and on top of that, it's the `.btn` class that defines the `--bs-btn-color` so overrides any value we'd set in `:root`.

### Reduced test cases

Run the documentation with `npm run docs:serve` and navigate to http://localhost:9001/docs/5.3/components/buttons/. The first example showing the base button will have no visible text.

### What operating system(s) are you seeing the problem on?

Linux

### What browser(s) are you seeing the problem on?

Firefox

### What version of Bootstrap are you using?

5.3

### PR DESCRIPTION
**Pull request by lavsemen**
### Description

Fix button color change issue on switching to dark theme: Resolved an issue where the button color was not changing properly on switching to the dark theme.

### Type of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [x] My code follows the code style of the project _(using `npm run lint`)_
- [x] My change introduces changes to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### Related issues

Closes https://github.com/twbs/bootstrap/issues/38113

### CODE REVIEW

File: scss/_buttons.scss [hunk 1]
```diff
@@ -10,7 +10,7 @@
   @include rfs($btn-font-size, --#{$prefix}btn-font-size);
   --#{$prefix}btn-font-weight: #{$btn-font-weight};
   --#{$prefix}btn-line-height: #{$btn-line-height};
-  --#{$prefix}btn-color: #{$body-color};
+  --#{$prefix}btn-color: var(--#{$prefix}body-color);
```
Comment (louismaximepiton) [inline scss/_buttons.scss:13]: We should maybe introduce a Sass variable (`$btn-color`) here for better customization.
This variable could be set either to `null` or as just mentionned `var(--#{$prefix}body-color)`.

**Review by louismaximepiton — COMMENTED** [review]
Hey, nice catch! 

Just a small enhancement and I think we're good to go. Thanks a lot for your PR 😄

**Comment by lavsemen** [comment]
@louismaximepiton  Moved color to sass variable!

File: scss/_variables.scss [hunk 2]
```diff
@@ -801,6 +801,7 @@ $input-btn-border-width:      var(--#{$prefix}border-width) !default;
 // For each of Bootstrap's buttons, define text, background, and border color.
 
 // scss-docs-start btn-variables
+$btn-color:                   var(--#{$prefix}body-color)!default;
```
Comment (mdo) [inline scss/_variables.scss:804]:
```suggestion
$btn-color:                   var(--#{$prefix}body-color) !default;
```

File: scss/_buttons.scss [hunk 3]
```diff
@@ -7,10 +7,10 @@
   --#{$prefix}btn-padding-x: #{$btn-padding-x};
   --#{$prefix}btn-padding-y: #{$btn-padding-y};
   --#{$prefix}btn-font-family: #{$btn-font-family};
-  @include rfs($btn-font-size, --#{$prefix}btn-font-size);
+   @include rfs($btn-font-size, --#{$prefix}btn-font-size);
```
Comment (mdo) [inline scss/_buttons.scss:10]:
```suggestion
  @include rfs($btn-font-size, --#{$prefix}btn-font-size);
```

**Review by mdo — APPROVED** [review]
Two quick changes I'll make and then good to go, thanks!

**Pull request merged**